### DISCUSSION TITLE
ESP32 littlefs data

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_midi_sampler#17] [Q&A]
Hey, can't understand what is 'littlefs data upload'.I have just bought an esp32 audio kit and want to make this project. can anyone guide me through?

**Comment by marcel-licence** [comment]
https://www.youtube.com/results?search_query=littlefs+tutorial there are a lot of tutorials showing how to upload data using the littlefs data upload. This is a little plugin for arduino which allows to transfer files in the data folder to your esp32. Hope this helps

**Reply by souravlal** [reply]
![IMG20210718125814](https://user-images.githubusercontent.com/53053463/126059547-e3a39017-32c2-4da1-9187-1385c14c3ffe.jpg)
Am I got it right 👆
But I can only see 'esp32 littlefs data upload' option in tools menu can't see 'Esp32 sketch upload'.

**Reply by marcel-licence** [reply]
Hm I've never seen this problem. Looks like arduino does not work as expected. Maybe a general arduino forum can help or a reinstall may be required? I have no idea sorry.

**Reply by souravlal** [reply]
Is there a problem?
I followed the procedure in the github Page.

**Reply by jazzbiddy** [reply]
@souravlal This guy has a good video on the setup of LittleFS: https://www.youtube.com/watch?v=4r6YZlLfKfw
Once you get it showing up properly - you need to remember to do the initial `ESP32 Sketch Data Upload" from that tools menu using the IDE, otherwise you'll see errors in the output about the mount not working. That step stumped me for a bit.

**Reply by marcel-licence** [reply]
@souravlal I am just wondering because there is no "ESP32 Sketch Data Upload" in your screenshot in your tools menu.

**Reply by souravlal** [reply]
![IMG20210720115914](https://user-images.githubusercontent.com/53053463/126275352-4571208f-660d-40d2-925c-bf1b4592e2f2.jpg)
Am I got iit Right this time?

**Reply by souravlal** [reply]
Are my settings correct? Do I need to click 'ESP32 sketch data upload' after connecting my board, or only click upload icon like we generally upload program to esp32?
Please help me.

**Reply by jazzbiddy** [reply]
@souravlal It looks like you have an empty / new sketch in the background. You should download Marcel's project and open "esp32_midi_sampler.ino" in the Arduino IDE. When the project is open - click on the "ESP32 Sketch Data Upload" item in the tools menu. This will upload the samples in the Data folder from the project. You only need to do this when you want to upload the samples from the data folder to the esp32 Audio Kit. After you've done that, then you should be able to do normal uploads to the board.

**Reply by souravlal** [reply]
Ok , thanks for the reply.
Are my settings correct? Like the board , flash mode, partition scheme, etc...?

**Reply by souravlal** [reply]
![esp32-SPIFFS-LittleFS-FatFS-file-uploader-from-Arduino-IDE](https://user-images.githubusercontent.com/53053463/126325005-5005bf46-c7a8-4cdc-a6b4-47e5be667f22.jpg)

When I click 'esp32 sketch data upload' it opens a window  like this 👆.  Which one I should choose?

**Reply by marcel-licence** [reply]
LittleFS should be the correct selection. I've never seen this menu. Looks like the sketch upload is flexible. I've just executed two steps -> running the normal upload using the "right arrow" button and after a manual "ESP32 LittleFS Data Upload".

The rest looks like this in my arduino:
![grafik](https://user-images.githubusercontent.com/65245426/126358909-b7cba5ff-afc6-4cae-810b-64a28b0067b6.png)
